src/plugins/lightscript.js

```diff
@@ -386,17 +386,36 @@ pp.parseNamedArrowFromCallExpression = function (node, call) {
   return this.finishNode(node, isMember ? "NamedArrowMemberExpression" : "NamedArrowExpression");
 };
 
+pp.pushBlockState = function (blockType, indentLevel) {
+  this.state.blockStack = this.state.blockStack || [];
+  this.state.blockStack.push({ blockType, indentLevel });
```

Comment:
have you played with try/catch at all yet? would rather just track indentLevel in something like `this.state.ifBlockStack` instead of an object if we're only using one blockType.